### ISSUE TITLE
BP Docs: Wrong doc is being deleted

### ISSUE THREAD

**Issue by bree-z** [cuny-academic-commons/commons-in-a-box#196]
Hi Boone, 

In troubleshooting a [question on the CBOX OL forum](https://commonsinabox.org/groups/openlab-help-support/forum/topic/group-docs-for-a-course-home-page-disappeared/) about a mysteriously deleted doc, I found a strange bug, which I’m not sure is actually related to her issue, but here’s what I found:

* I created a doc here: https://openlabdev.commonsinabox.org/groups/writing-politics-sp2018-pol345/docs/ called “Doc 1 - Writing Politics”.
* I created another doc called “Doc 2 - Writing Politics,” which I deleted on the [main Docs page](https://openlabdev.commonsinabox.org/groups/writing-politics-sp2018-pol345/docs/), by mousing over the title and clicking delete.  This method of deleting docs seems to be fine - the correct one is deleted.
* I created a third doc called “Doc 3 - Writing Politics”.  I clicked on the title, then clicked edit, and then clicked the delete link at the bottom.  I ended up on the main Docs page again, but “Doc 3” was still there and “Doc 1” had been deleted.  

I saw the same behavior in another group where I was also testing: https://openlabdev.commonsinabox.org/groups/ancient-political-thought/docs/

Here's a gif to show the wrong doc being deleted:

![DocsDeletion3](https://user-images.githubusercontent.com/7189824/66783911-06a5f780-eea7-11e9-96d0-e61f04da383a.gif)

Thanks!

**Comment by boonebgorges** [comment]
@bree-z Thank you very much for the very thorough report. I've confirmed the behavior.

Briefly, this is another recurrence of a Docs bug discussed in https://github.com/boonebgorges/buddypress-docs/issues/612. CBOX-OL uses the buddypress-docs-in-group plugin to put Docs into the group interface. This change requires a fair amount of juggling, since the normal logic used by Docs to determine the "current" Doc is tied to WP's URL routing system; when we're in a group, we've got to override all of this logic. In this ticket, you've identified another place in Docs where the value was not correctly filterable. This is fixed in https://github.com/boonebgorges/buddypress-docs/issues/661 and will be part of the next Docs and CBOX-OL release. In the near term, administrators can manually patch their installations: https://github.com/boonebgorges/buddypress-docs/commit/91509606cc9ebaa1fbbbbb02abb2bbca5c8188cb

A related UX issue is that the Delete button was visible during the Doc creation process. This is semantically incorrect, and clicking the button in this context could have weird consequences. I've just pushed up a change to openlab-theme that prevents the button from showing during Doc creation (still appears during Edit mode, of course).

These changes are available on openlabdev.commonsinabox.org for verification.
